pkg/services/live/pipeline/output_changelog.go

```diff
@@ -0,0 +1,92 @@
+package pipeline
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+type ChangeLogOutputConfig struct {
+	FieldName string `json:"fieldName"`
+	Channel   string `json:"channel"`
+}
+
+// ChangeLogOutput can monitor value changes of the specified field and output
+// special change frame to the configured channel.
+type ChangeLogOutput struct {
+	frameStorage FrameGetSetter
+	config       ChangeLogOutputConfig
+}
+
+func NewChangeLogOutput(frameStorage FrameGetSetter, config ChangeLogOutputConfig) *ChangeLogOutput {
+	return &ChangeLogOutput{frameStorage: frameStorage, config: config}
+}
+
+func (l ChangeLogOutput) Output(_ context.Context, vars OutputVars, frame *data.Frame) ([]*ChannelFrame, error) {
+	previousFrame, previousFrameOK, err := l.frameStorage.Get(vars.OrgID, l.config.Channel)
+	if err != nil {
+		return nil, err
+	}
+
+	fieldName := l.config.FieldName
+
+	previousFrameFieldIndex := -1
+	if previousFrameOK {
+		for i, f := range previousFrame.Fields {
+			if f.Name == fieldName {
+				previousFrameFieldIndex = i
+			}
+		}
+	}
+
+	currentFrameFieldIndex := -1
+	for i, f := range frame.Fields {
+		if f.Name == fieldName {
+			currentFrameFieldIndex = i
+		}
+	}
+
+	var previousValue interface{}
+	if previousFrameFieldIndex >= 0 {
+		// Take last value for the field.
+		previousValue = previousFrame.Fields[previousFrameFieldIndex].At(previousFrame.Fields[previousFrameFieldIndex].Len() - 1)
+	}
+
+	fTime := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
```

Comment:
Just to point out it looks like a good opportunity to add a more helpful func to the SDK

Reply:
For the future ✨

Reply:
Yeah, and actually helpers like GetFieldIndexByName.